contractcourt/htlc_outgoing_contest_resolver_test.go

```diff
@@ -73,6 +75,96 @@ func TestHtlcOutgoingResolverRemoteClaim(t *testing.T) {
 	ctx.waitForResult(false)
 }
 
+// TestHtlcOutgoingResolverRemoteClaim tests resolution of an offered htlc that
+// is claimed by the remote party. Specifically, claiming by sending a
+// retribution transaction is tested.
+func TestHtlcOutgoingResolveWithFailure(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	tests := []struct {
+		name         string
+		isEarlySpend bool
+	}{
+		{
+			name:         "output spent before resolver started",
+			isEarlySpend: true,
+		},
+		{
+			name:         "output spent after resolver started",
+			isEarlySpend: false,
+		},
+	}
+
+	for _, test := range tests {
+		// Setup the resolver with our test resolution and start the
+		// resolution process.
+		ctx := newOutgoingResolverTestContext(t)
+
+		// The remote party sweeps the HTLC by sending a retribution
+		// transaction. Notify our resolver of this event.
+		revokKey := bytes.Repeat([]byte{0x00}, 33)
+		spendDetails := &chainntnfs.SpendDetail{
+			SpendingTx: &wire.MsgTx{
+				TxIn: []*wire.TxIn{
+					{
+						Witness: [][]byte{
+							{0}, revokKey},
+					},
+				},
+			},
+		}
+
+		if test.isEarlySpend {
+			// To simulate an early spend, we send spendDetails
+			// before running the resolver. Also, the resolver
+			// shouldn't expect for new block epochs.
+			go func() {
+				ctx.notifier.spendChan <- spendDetails
+			}()
+			ctx.resolve(!test.isEarlySpend)
+		} else {
+			// To simulate a late spend, we send spendDetails after
+			// running the resolver. The resolver also receives a
+			// new block epoch.
+			ctx.resolve(test.isEarlySpend)
+			ctx.notifier.spendChan <- spendDetails
+		}
+
+		// We expect a resolution message to the incoming side of the
+		// circuit.
+		resMsg := <-ctx.resolutionChan
+
+		// We expect preImage to be nil because it wasn't revealed.
+		if resMsg.PreImage != nil {
```

Comment:
Optionally, you can use the newly introduced testify functions here, like `require.NotNil(resMsg.PreImage)`